packages/flutter/lib/src/widgets/platform_view.dart

```diff
@@ -358,47 +360,296 @@ class AppKitView extends _DarwinView {
+/// * Read more about [`ResizeObserver` in the MDN](https://developer.mozilla.org/en-US/docs/Web/API/Resize_Observer_API).
+///
+/// #### Other Observers
+///
+/// The example above uses a `ResizeObserver` because it can be applied directly
+/// to the `element` that is about to be injected. The most _correct_ observer
+/// for this (albeit requiring slightly more code) would be a
+/// [`MutationObserver`](https://developer.mozilla.org/en-US/docs/Web/API/MutationObserver).
+///
+/// The `MutationObserver` requires the "parent" element in which the
+/// `HtmlElementView` is going to be inserted. A safe way to retrieve a "parent"
+/// element for the platform views is to retrieve the `hostElement` of the
+/// [FlutterView] where the `HtmlElementView` is being rendered.
+///
+/// The `hostElement` of the current [FlutterView] can be retrieved through:
+///
+/// ```dart
+/// import 'dart:js_interop';
+/// import 'dart:ui_web' as ui_web;
+/// import 'package:flutter/widgets.dart';
+///
+/// void useHostElement(BuildContext context) {
+///   final int flutterViewId = View.of(context).viewId;
+///   final JSAny? hostElement = ui_web.views.getHostElement(flutterViewId);
+///   // Use `package:web` with `hostElement`...
+/// }
+/// ```
+///
+/// **Important:** If your flutter App uses `registerViewFactory` to register a
+/// factory function that creates your `HtmlElementView` widget, take into
+/// account that that the `viewId` parameter passed to your factory function
+/// **is not the same** as the `flutterViewId` obtained above:
+///
+/// * `flutterViewId` (from `View.of(context)`) represents the [FlutterView]
+///   where the web app is currently rendering.
+/// * `viewId` (passed to the `viewFactory` function) represents a unique ID
+///   for the `HtmlElementView` instance that is being injected into the app.
+///
+/// Read more about [FlutterView] on Flutter's API docs:
+///
+/// * [`View.of`](https://api.flutter.dev/flutter/widgets/View/of.html)
+/// * [`getHostElement`](https://main-api.flutter.dev/flutter/dart-ui_web/FlutterViewManagerProxy/getHostElement.html)
+///
+/// ## Pointer events
+///
+/// `HtmlElementView` will interfere with Flutter's normal gesture detection. In
+/// order for the `HtmlElementView` contents to be interactive, they're allowed
+/// to handle `pointer-events`. This may result in Flutter missing some events
+/// because they've been handled by the `HtmlElementView`, and never seen by
+/// Flutter.
+///
+/// [`package:pointer_interceptor`](https://pub.dev/packages/pointer_interceptor)
+/// may help in some cases where Flutter content needs to be overlaid on top of
+/// an `HtmlElementView`. Alternatively, the `pointer-events: none` property can
+/// be set `onElementCreated`; but that will prevent **ALL** interactions with
+/// the underlying HTML content.
+///
+/// If the `HtmlElementView` is an `<iframe>` element, Flutter will not be able
+/// to see any pointer events that land in the `<iframe>` (click/tap, drag,
```

Comment:
nit: replace "not be able to see any" with "receive". BTW, is this always true? Can't you create an iframe with origin trust policies that allow observing cross-frame events?

Reply:
> Can't you create an iframe with origin trust policies that allow observing cross-frame events?

Technically, if Flutter web could see the `<iframe>`'s [`contentWindow` property](https://developer.mozilla.org/en-US/docs/Web/API/HTMLIFrameElement/contentWindow), we could `iframe.contentWindow.addEventListener('pointermove'...` (for example) and try to merge events from the iframe with those that are reported to the framework.

In practice, we don't do this in our pointer handling (and I'm not sure if there's a way for users to funnel events from the app to the framework), so the answer to your question is: yes, _with our current implementation_, this is always true.

Reply:
(Also, rephrased the other bits)